pkg/code-generator/docgen/generator.go

```diff
@@ -147,22 +147,37 @@ func (d *DocsGen) FileHeader(filename string) string {
 		return ".. Code generated by solo-kit. DO NOT EDIT."
 	}
 	if d.DocsOptions.Output == options.Hugo {
-		name := filepath.Base(filename)
-		if strings.HasSuffix(name, d.protoSuffix()) {
-			name = name[:len(name)-len(d.protoSuffix())]
-		}
-		return fmt.Sprintf(`
+		return d.hugoFileHeader(filename)
+	}
+	return `<!-- Code generated by solo-kit. DO NOT EDIT. -->
+`
+}
+
+func (d *DocsGen) hugoFileHeader(filename string) string {
+	name := filepath.Base(filename)
+
+	var title string
+	if strings.HasSuffix(name, d.protoSuffix()) {
+		// Remove the "sk.md" extensions
+		name = name[:len(name)-len(d.protoSuffix())]
+
+		// Remove the ".proto" extension
+		extension := filepath.Ext(name)
+		title = strcase.ToCamel(name[0 : len(name)-len(extension)])
```

Comment:
I noticed in the `gloo` PR this results in these titles: `GatewaySoloIo`, `EnvoyGlooeSoloIo`
Is that what we want?
Perhaps we only want to camelCase names that have no `.` remaining after stripping the extension?

Reply:
so I agree with @arianaw66  that in this case the name does not make sense. Only the ones that end in `.proto` I think should get the name change. the ones that end in `.project` should not. However, I also think we should not have any of these .project resources in our docs, especially since those are just overview pages with all the CRs. So what we could do is one of the following: 

Option 1: 
- Not change the name for .project files. 
- Display them as-is in the docs. 

Option 2: 
- Leave the change as-is and remove them from the docs. We can do this manually on our end if it can't be done in the template or is too hard to implement.